stats.c

```
@@ -2426,80 +2433,45 @@ int main_stats(int argc, char *argv[])
+            if (iter) {
+                if (!targets) {
+                    all_stats->nchunks = argc-optind;
+                    if (replicate_regions(all_stats, iter))
+                        fprintf(stderr, "Replications of the regions failed\n");
+                }
 
-                        if ( all_stats->nregions && all_stats->regions ) {
-                            while (sam_itr_multi_next(info->sam, iter, bam_line) >= 0) {
-                               if (info->split_tag) {
-                                   curr_stats = get_curr_split_stats(bam_line, split_hash, info, targets);
-                                   collect_stats(bam_line, curr_stats, read_pairs);
-                               }
-                               collect_stats(bam_line, all_stats, read_pairs);
-                            }
+                if ( all_stats->nregions && all_stats->regions ) {
+                    while (sam_itr_next(info->sam, iter, bam_line) >= 0) {
+                        if (info->split_tag) {
+                            curr_stats = get_curr_split_stats(bam_line, split_hash, info, targets);
+                            collect_stats(bam_line, curr_stats, read_pairs);
                         }
-
-                        hts_itr_multi_destroy(iter);
-                    } else {
-                       fprintf(stderr, "Creation of the region iterator failed.");
-                       hts_reglist_free(reglist, regcount);
+                        collect_stats(bam_line, all_stats, read_pairs);
                     }
-                } else {
-                    fprintf(stderr, "Creation of the region list failed.");
                 }
-
-                hts_idx_destroy(bam_idx);
+                hts_itr_multi_destroy(iter);
             } else {
-                fprintf(stderr, "Random alignment retrieval only works for indexed BAM files.\n");
+                fprintf(stderr, "Multi-region iterator could not be created\n");
```

Probably want to return an error here